src/Server.ts

```diff
@@ -1,46 +1,59 @@
 import * as bodyParser from 'body-parser';
 import * as cookieParser from 'cookie-parser';
 import * as express from 'express';
-import * as session from 'express-session';
 import * as nunjucks from 'nunjucks';
+import 'reflect-metadata';
 import * as path from 'path';
-import { router } from './routes/Routes';
+import { Container } from 'inversify';
+import { InversifyExpressServer } from 'inversify-express-utils';
+import './controllers/index';
+import { RedisService } from './services/RedisService';
 
 export class Server {
 
-  private app: express.Express;
+  private server: InversifyExpressServer;
+  private port: number;
+
+  constructor(port: number) {
+    this.port = port;
+    this.server = new InversifyExpressServer(this.setupBindings());
+    this.server.setConfig((app) => {
+      app.set('port', port);
+      this.setupStaticFolders(app);
+      this.setupParsers(app);
+      this.setViewEngine(app);
+    });
 
-  constructor(port?: number) {
-    this.app = express();
-    this.app.set('port', process.env.PORT || port || 3000);
-    this.setupStaticFolders(this.app);
-    this.setupParsers(this.app);
-    this.setViewEngine(this.app);
-    this.setupAppConfig(this.app);
   }
 
   public start(): void {
-    this.app.listen(this.app.get('port'), () => {
-      console.log(('  App is running at http://localhost:%d in %s mode'), this.app.get('port'), this.app.get('env'));
+    this.server.build().listen(this.port, () => {
+      console.log(('  App is running at http://localhost:%d in %s mode'), this.port, process.env.NODE_ENV);
       console.log('  Press CTRL-C to stop\n');
     });
   }
 
-  private setupStaticFolders(app: express.Express): void {
+  private setupBindings(): Container {
```

Comment:
I feel like this method name is a little misleading, it suggests we are only setting up bindings, but we are actually creating a new container, configuring it's bindings and then returning the container. Instead we could create the Container outside of the method and pass it in as a parameter?

Reply:
I'll change this to `createContainerWithBindings(): Container` because creating a container outside and passing it in as a parameter only to then mutate it's state is a side effect we can avoid ;)